code/modules/antagonists/pirate/pirate_gangs.dm

```diff
@@ -0,0 +1,128 @@
+	///the key to the json list of pirate names
+	var/ship_name_pool = "some_json_key"
+	///inbound message title the station recieves
+	var/threat_title = "Pay away the Space Bugs"
+	///the contents of the message sent to the station.
+	///%SHIPNAME in the content will be replaced with the pirate ship's name
+	///%PAYOFF in the content will be replaced with the requested credits.
+	var/threat_content = "This is the %SHIPNAME. Give us %PAYOFF credits or we bug out the universe trying to spawn!"
+	///what the station can say in response, first item pays the pirates, second item rejects it.
+	var/list/possible_answers = list("Please, go away! We'll pay!", "I accept oblivion.")
+
+	///station responds to message and pays the pirates
+	var/response_recieved = "Yum! Bungopoints!"
+	///station pays the pirates, but after the ship spawned
+	var/response_too_late = "Your Bungopoints arrived too late, rebooting world..."
+	///station pays the pirates... but doesn't have enough cash.
+	var/response_not_enough = "Not enough Bungopoints have been added into my bank account, rebooting world..."
+
+/datum/pirate_gang/New()
+	. = ..()
+	ship_name = pick(strings(PIRATE_NAMES_FILE, ship_name_pool))
+
+///whether this pirate gang can roll today. this is called when the global list initializes, so
+///returning FALSE means it cannot show up at all for the entire round.
+/datum/pirate_gang/proc/can_roll()
+	return TRUE
+
+///returns a new comm_message datum from this pirate gang
+/datum/pirate_gang/proc/generate_message(payoff)
+	var/built_threat_content = replacetext(threat_content, "%SHIPNAME", ship_name)
+	built_threat_content = replacetext(threat_content, "%PAYOFF", payoff)
+	return new /datum/comm_message(threat_title, built_threat_content, possible_answers)
+
+///classic FTL-esque space pirates.
+/datum/pirate_gang/rogues
+	name = "Rogues"
+
+	ship_template = /datum/map_template/shuttle/pirate/default
+	ship_name_pool = "rogue_names"
+
+	threat_title = "Sector protection offer"
+	threat_content = "Hey, pal, this is the %SHIPNAME. Can't help but notice you're rocking a wild \
+		and crazy shuttle there with NO INSURANCE! Crazy. What if something happened to it, huh?! We've \
+		done a quick evaluation on your rates in this sector and we're offering %PAYOFF to cover for your \
+		shuttle in case of any disaster."
+	possible_answers = list("Purchase Insurance.","Reject Offer.")
+
+	response_recieved = "Sweet, free cash. Let's get outta here, boys."
+	response_too_late = "Payment or not, ignoring us was a matter of pride. Now it's time for us to teach some respect."
+	response_not_enough = "You thought we wouldn't notice if you underpaid? Funny. We'll be seeing you soon."
+
+///aristocrat lizards looking to hunt the serfs
+/datum/pirate_gang/silverscales
+	name = "Silverscales"
+
+	ship_template = /datum/map_template/shuttle/pirate/silverscale
+	ship_name_pool = "silverscale_names"
+
+	threat_title = "Sector protection offer"
```

Comment:
doesnt fit, neither does the skeleton one

Reply:
ty